src/contexts/SolanaWalletContext.tsx

```diff
@@ -47,8 +79,11 @@ export const SolanaWalletProvider: FC = (props) => {
     if (network) {
       wallets.push(new BloctoWalletAdapter({ network }));
     }
+
     return wallets;
-  }, []);
+  }, [isBrave]);
+
+  if (isBrave === null) return <></>;
```

Comment:
Is this check necessary?

Reply:
Yes, if you do not include that check, it looks like the `WalletProvider` only runs once with the first `wallets` array and does not react to the next iteration.